GunShipAlpha/src/Enemy.java

```diff
@@ -1,41 +1,94 @@
 import java.awt.*;
+import java.awt.geom.Point2D;
+import java.util.Random;
+import java.util.Timer;
+import java.util.TimerTask;
 
 class Enemy {
-    private int x, y;
-    private final int targetX, targetY;
+    private Point location;
+    private Point target;
     private int health = 2;
+    private Timer firingTimer;
+
     public Enemy(int x, int y, int targetX, int targetY) {
-        this.x = x;
-        this.y = y;
-        this.targetX = targetX;
-        this.targetY = targetY;
+        this.location = new Point(x, y);
+        this.target = new Point(targetX, targetY);
+        scheduleFiring();
     }
+    private void scheduleFiring() {
+        if (firingTimer != null) {
+            firingTimer.cancel();
+        }
+
+        firingTimer = new Timer();
+
+        TimerTask firingTask = new TimerTask() {
+            @Override
+            public void run() {
 
+                Projectile projectile = fireAtPlayer();
+                Game.gs.addProjectile(projectile);
+                scheduleFiring();
+            }
+        };
+        Random random = new Random();
+        long delay = 0 + random.nextInt(3000);
+        firingTimer.schedule(firingTask, delay);
+    }
     public void moveTowardTarget() {
-        if (x < targetX) x++;
-        if (x > targetX) x--;
-        if (y < targetY) y++;
-        if (y > targetY) y--;
+        if (hasEnemyReachedTarget()) return;
+        if (location.x < target.x) location.x++;
+        if (location.x> target.x) location.x--;
+        if (location.y < target.y) location.y++;
+        if (location.y > target.y) location.y--;
+    }
+
+    private boolean hasEnemyReachedTarget() {
+        if(location.x == target.x && location.y == target.y ) {
+            health-=2;
+            Game.gs.removeEnemy();
+            return true;
+        }
+        return false;
     }
+
     void removeHealth() {
         health--;
+        if (isDead()) {
+            firingTimer.cancel();
+        }
     }
     public boolean isCollidingWithProjectile(Projectile projectile) {
-        Rectangle enemyBounds = new Rectangle(x, y, EnemySprite.width(), EnemySprite.height());
+        Rectangle enemyBounds = EnemySprite.getBounds(location.x, location.y);
```

Comment:
just pass location. the point of (dang, pun again) having the point object is to reduce the number of places you have to deal with x and y to the minimum.

Reply:
im having a hard time understanding exactly what you mean by this.